fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/InclusionQuerySegmentAggregator.java

```diff
@@ -199,23 +199,41 @@ private void processIncludeParameters(StringBuilder queryString, List<Object> bi
             // R.RESOURCE_ID = LR.CURRENT_RESOURCE_ID AND
             queryString.append("R.RESOURCE_ID = LR.CURRENT_RESOURCE_ID AND ");
             // ('Organization/' || LR.LOGICAL_ID IN 
-            queryString.append("('").append(includeParm.getSearchParameterTargetType()).append("/' || LR.LOGICAL_ID IN ");
-            // (SELECT P1.STR_VALUE FROM OBSERVATION_STR_VALUES P1 WHERE
-            queryString.append("(SELECT P1.STR_VALUE FROM ").append(this.resourceType.getSimpleName()).append("_STR_VALUES P1 WHERE ");
+            queryString.append("('").append(includeParm.getSearchParameterTargetType()).append("/' || LR.LOGICAL_ID IN ("); 
+
+            // Run the following sub Query only once to improve performance.
```

Comment:
but its in a FOR loop...

is this comment saying the code should be improved to only execute a sub-query once?

Reply:
the subquery gives string values for IN

Reply:
maybe change it to something like this?
```suggestion
            // This section purposefully breaks from the pattern of constructing a single query and going to the database just once.
            // We found that, for include queries, this was actually contributing to performance issues due to large temporary table management,
            // and so now this subquery is executed independently for each inclusion parameter.
```

Reply:
let me make it more clear in the comments, this prevent the subquery to be executed once for each records of the joined table.

Reply:
thanks.  I think we be very explicit about the fact we're executing a separate query here because, like i mention in my suggested comment, it breaks from the pattern used in the rest of the QueryBuilder

Reply:
done